packages/react-devtools-shared/src/devtools/views/Profiler/ProfilerContext.js

```diff
@@ -129,6 +129,38 @@ function ProfilerContextController({children}: Props) {
     setPrevProfilingData,
   ] = useState<ProfilingDataFrontend | null>(null);
   const [rootID, setRootID] = useState<number | null>(null);
+  const [selectedFiberID, selectFiberID] = useState<number | null>(null);
+  const [selectedFiberName, selectFiberName] = useState<string | null>(null);
+
+  const selectFiber = useCallback(
+    (id: number | null, name: string | null) => {
+      selectFiberID(id);
+      selectFiberName(name);
+
+      // Sync selection to the Components tab for convenience.
+      if (id !== null) {
+        const element = store.getElementByID(id);
+
+        // Keep in mind that profiling data may be from a previous session.
+        // In that case, IDs may match up arbitrarily; to be safe, compare both ID and display name.
```

Comment:
Ah, right. So this seems to come up in a few places. I know this check was copied from pre-existing code, but looking at it again- the heuristic is flawed. ID and name might still happen to match for imported data.

We should check something on the data itself to show that it was imported or not, and just skip looking in the store for such imported elements.

We could add a new field to the `ProfilingDataFrontend` type to track this. Interested in tackling this? (Totally fine if you aren't, I could do it.)

Reply:
Yeah, it could be interesting to do. Should this field be committed to this PR, the task is all in this comment?

Reply:
Just to be clear, this doesn't *need* to be done- at least not as part of this PR.

Reply:
I'll try to do it in the next few days, thanks!

Reply:
https://github.com/facebook/react/pull/18880